tools/plugins/com.liferay.ide.upgrade.tasks.core/src/com/liferay/ide/upgrade/tasks/core/internal/code/ImportLiferayWorkspaceAction.java

```diff
@@ -39,7 +47,24 @@
 
 	@Override
 	public IStatus perform(IProgressMonitor progressMonitor) {
-		return Status.OK_STATUS;
+		Path path = _resourceSelection.selectPath("Please select the workspace location.");
+
+		if (path == null) {
+			return Status.CANCEL_STATUS;
+		}
+
+		LiferayGradleWorkspaceProjectProvider provider = new LiferayGradleWorkspaceProjectProvider();
```

Comment:
This should not be done here to add a dependency to LiferayGradleWorkspaceProjectProvider and instead we should inject a ProjectImportor that can handle "workspace". and we can register a service implementaiton from gradle.core plugin.

Reply:
got it, will do it